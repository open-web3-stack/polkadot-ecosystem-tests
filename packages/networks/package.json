{
	"name": "@e2e-test/networks",
	"type": "module",
	"dependencies": {
		"@acala-network/chopsticks-testing": "^0.13.1",
		"@e2e-test/shared": "workspace:*"
	},
	"main": "./dist/cjs/index.js",
	"module": "./dist/esm/index.js",
	"files": [
		"*",
		"!tsconfig.json",
		"!**/*.test.ts"
	],
	"exports": {
		".": {
			"require": "./dist/cjs/index.js",
			"import": "./dist/esm/index.js",
			"default": "./dist/esm/index.js"
		},
		"./chains": {
			"require": "./dist/cjs/chains/index.js",
			"import": "./dist/esm/chains/index.js",
			"default": "./dist/esm/chains/index.js"
		},
		"./*": {
			"require": "./dist/cjs/*.js",
			"import": "./dist/esm/*.js",
			"default": "./dist/esm/*.js"
		}
	}
}
