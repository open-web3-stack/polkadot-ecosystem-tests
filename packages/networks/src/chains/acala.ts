import { defaultAccount } from '@e2e-test/shared'

import { defineChain } from '../defineChain.js'

const custom = {
  acala: {
    relayToken: 'DOT',
    relayLiquidToken: 'LDOT',
    stableToken: 'AUSD',

    paraAccount: '13YMK2eYoAvStnzReuxBjMrAvPXmmdsURwZvc62PrdXimbNy',
    dot: { Token: 'DOT' },
    ldot: { Token: 'LDOT' },
    dai: { Erc20: '0x54a37a01cd75b616d63e0ab665bffdb0143c52ae' },
    wbtc: { ForeignAsset: 5 },
    ausd: { Token: 'AUSD' },
    aca: { Token: 'ACA' },
    lcdot: { LiquidCrowdloan: 13 },
  },
  karura: {
    relayToken: 'KSM',
    relayLiquidToken: 'LKSM',
    stableToken: 'KUSD',

    paraAccount: '13YMK2eYoAvStnzReuxBjMrAvPXmmdsURwZvc62PrdXimbNy',
    ksm: { Token: 'KSM' },
    lksm: { Token: 'LKSM' },
    usdt: { ForeignAsset: 7 },
    rmrk: { ForeignAsset: 0 },
    dai: { Erc20: '0x4bb6afb5fa2b07a5d1c499e1c3ddb5a15e709a71' },
    ausd: { Token: 'KUSD' },
    kar: { Token: 'KAR' },
  },
}

const getInitStorages = (config: typeof custom.acala | typeof custom.karura) => ({
  System: {
    account: [[[defaultAccount.alice.address], { providers: 4, data: { free: 10 * 1e12 } }]],
  },
  Tokens: {
    accounts: [
      [[defaultAccount.alice.address, { Token: config.relayToken }], { free: 10 * 1e12 }],
      [[defaultAccount.alice.address, { Token: config.relayLiquidToken }], { free: 100 * 1e12 }],
      [[defaultAccount.alice.address, { Token: config.stableToken }], { free: 1000 * 1e12 }],
    ],
  },
  Sudo: {
    key: defaultAccount.alice.address,
  },
  EvmAccounts: {
    accounts: [[['0x82a258cb20e2adb4788153cd5eb5839615ece9a0'], defaultAccount.alice.address]],
    evmAddresses: [[[defaultAccount.alice.address], '0x82a258cb20e2adb4788153cd5eb5839615ece9a0']],
  },
  Homa: {
    // avoid impact test outcome
    $removePrefix: ['redeemRequests', 'unbondings', 'toBondPool'],
    // so that bump era won't trigger unbond
    relayChainCurrentEra: 100,
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers', 'supportedVersion'],
  },
})

export const acala = defineChain({
  name: 'acala',
  endpoint: 'wss://acala-rpc.aca-api.network',
  paraId: 2000,
  custom: custom.acala,
  initStorages: getInitStorages(custom.acala),
})

export const karura = defineChain({
  name: 'karura',
  endpoint: 'wss://karura-rpc.aca-api.network',
  paraId: 2000,
  custom: custom.karura,
  initStorages: getInitStorages(custom.karura),
})
