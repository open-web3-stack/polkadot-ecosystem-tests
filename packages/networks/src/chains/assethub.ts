import { defaultAccounts } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  assetHubPolkadot: {
    dot: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
  assetHubKusama: {
    ksm: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
}

const getInitStorages = (config: typeof custom.assetHubPolkadot | typeof custom.assetHubKusama) => ({
  System: {
    account: [[[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }]],
  },
  Assets: {
    account: [
      [[config.usdtIndex, defaultAccounts.alice.address], { balance: 1000e6 }], // USDT
    ],
  },
})

export const assetHubPolkadot = defineChain({
  name: 'assetHubPolkadot',
  endpoint: 'wss://rpc-asset-hub-polkadot.luckyfriday.io',
  paraId: 1000,
  custom: custom.assetHubPolkadot,
  initStorages: getInitStorages(custom.assetHubPolkadot),
})

export const assetHubKusama = defineChain({
  name: 'assetHubKusama',
  endpoint: 'wss://rpc-asset-hub-kusama.luckyfriday.io',
  paraId: 1000,
  custom: custom.assetHubKusama,
  initStorages: getInitStorages(custom.assetHubKusama),
})
