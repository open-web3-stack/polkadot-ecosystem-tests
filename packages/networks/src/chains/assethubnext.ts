import { defaultAccounts, defaultAccountsSr25519 } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  assetHubNextWestend: {
    wnd: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
}

const getInitStorages = (config: typeof custom.assetHubNextWestend) => ({
  System: {
    account: [
      [[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }],
      [[defaultAccountsSr25519.alice.address], { providers: 1, data: { free: 1000e10 } }],
    ],
  },
  Assets: {
    account: [
      [[config.usdtIndex, defaultAccounts.alice.address], { balance: 1000e6 }], // USDT
    ],
  },
})

export const assetHubNextWestend = defineChain({
  name: 'assetHubNextWestend',
  endpoint: 'wss://westend-asset-hub-next-rpc.parity-chains-scw.parity.io',
  paraId: 1100,
  custom: custom.assetHubNextWestend,
  initStorages: getInitStorages(custom.assetHubNextWestend),
})
