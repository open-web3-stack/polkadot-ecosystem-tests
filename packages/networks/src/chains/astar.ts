import { defaultAccounts } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  astar: {
    relayToken: '340282366920938463463374607431768211455',
    aUSDToken: '18446744073709551617',

    paraAccount: '13YMK2eZzuFY1WZGagpYtTgbWBWGdoUD2CtrPj1mQPjY8Ldc',
    dot: 340282366920938463463374607431768211455n,
    astr: { Concrete: { parents: 0, interior: 'Here' } },
    aca: 18446744073709551616n,
    usdt: 4294969280n,
  },
  shiden: {
    relayToken: '340282366920938463463374607431768211455',
    aUSDToken: '18446744073709551616',

    paraAccount: 'F7fq1jNy74AqkJ1DP4KqSrWtnTGtXfNVoDwFhTvvPxUvJaq',
    ksm: 340282366920938463463374607431768211455n,
    sdn: { Concrete: { parents: 0, interior: 'Here' } },
    kar: 18446744073709551618n,
    usdt: 4294969280n,
  },
}

const getInitStorages = (config: typeof custom.astar | typeof custom.shiden) => ({
  System: {
    account: [[[defaultAccounts.alice.address], { providers: 1, data: { free: '100000000000000000000' } }]],
  },
  Assets: {
    account: [
      [[config.relayToken, defaultAccounts.alice.address], { balance: 10 * 1e12 }],
      [[config.aUSDToken, defaultAccounts.alice.address], { balance: 10 * 1e12 }],
    ],
  },
  Sudo: {
    key: defaultAccounts.alice.address,
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers', 'supportedVersion'],
  },
})

export const astar = defineChain({
  name: 'astar',
  paraId: 2006,
  endpoint: 'wss://astar-rpc.dwellir.com',
  custom: custom.astar,
  initStorages: getInitStorages(custom.astar),
})

export const shiden = defineChain({
  name: 'shiden',
  paraId: 2007,
  endpoint: 'wss://shiden-rpc.dwellir.com',
  custom: custom.shiden,
  initStorages: getInitStorages(custom.shiden),
})
