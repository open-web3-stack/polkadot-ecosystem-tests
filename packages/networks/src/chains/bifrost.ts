import { defineChain } from '../defineChain.js'
import { defaultAccounts, defaultAccountsSr25519, testAccounts } from '../testAccounts.js'

const custom = {
  bifrostPolkadot: {
    relayToken: 'DOT',
    bnc: { Native: 'BNC' },
  },
  bifrostKusama: {
    relayToken: 'KSM',
    bnc: { Token: 'BNC' },
  },
}

const getInitStorages = (_config: typeof custom.bifrostPolkadot | typeof custom.bifrostKusama) => ({
  System: {
    account: [
      [[defaultAccounts.alice.address], { providers: 1, data: { free: 666e12 } }],
      [[defaultAccountsSr25519.alice.address], { providers: 1, data: { free: 666e12 } }],
      [[testAccounts.alice.address], { providers: 1, data: { free: 666e12 } }],
    ],
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers'],
  },
})

export const bifrostPolkadot = defineChain({
  name: 'bifrostPolkadot',
  endpoint: 'wss://bifrost-polkadot.ibp.network',
  paraId: 2030,
  custom: custom.bifrostPolkadot,
  initStorages: getInitStorages(custom.bifrostPolkadot),
})

export const bifrostKusama = defineChain({
  name: 'bifrostKusama',
  endpoint: 'wss://us.bifrost-rpc.liebi.com/ws',
  paraId: 2001,
  custom: custom.bifrostKusama,
  initStorages: getInitStorages(custom.bifrostKusama),
})
