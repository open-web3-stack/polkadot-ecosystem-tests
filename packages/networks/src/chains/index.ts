export * from './acala.js'
export * from './assethub.js'
export * from './astar.js'
export * from './hydraDX.js'
export * from './moonbeam.js'
export * from './polkadot.js'
