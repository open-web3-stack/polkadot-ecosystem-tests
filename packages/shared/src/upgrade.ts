import { sendTransaction } from '@acala-network/chopsticks-testing'

import { type Chain, type Client, defaultAccounts } from '@e2e-test/networks'
import { sendWhitelistCallViaXcmTransact } from '@e2e-test/shared'

import type { HexString } from '@polkadot/util/types'

import { assert } from 'vitest'

import { checkEvents, checkSystemEvents, createXcmTransactSend, scheduleInlineCallWithOrigin } from './helpers/index.js'

/**
 * Computes the XCM `MultiLocation` route from a source chain to a destination chain.
 *
 * @param from - The source chain (the chain initiating the XCM message).
 * @param to - The destination chain (the chain intended to receive and execute the XCM message).
 */
function getXcmRoute(from: Chain, to: Chain) {
  let parents: number
  let interior: any

  if (from.isRelayChain) {
    parents = 0
  } else {
    parents = 1
  }

  if (to.isRelayChain) {
    interior = 'Here'
  } else {
    interior = { X1: [{ Parachain: to.paraId }] }
  }

  return { parents, interior }
}

/**
 * Constructs an XCM forceBatch transaction that authorizes a runtime upgrade on a destination chain.
 *
 *
 * @param codeHash - The code hash of the new runtime to be authorized.
 * @param sourceClient - The client instance representing the source chain (where the XCM is sent from).
 * @param destClient - The client instance representing the destination chain (where the upgrade is authorized).
 */
export function createXcmAuthorizeUpgradeBatch(codeHash: HexString, sourceClient: Client, destClient: Client) {
  const authorizeUpgradeCall = destClient.api.tx.system.authorizeUpgrade(codeHash)
  const dest = getXcmRoute(sourceClient.config, destClient.config)

  const xcmTx = createXcmTransactSend(sourceClient, dest, authorizeUpgradeCall.method.toHex(), 'Superuser', {
    refTime: '5000000000',
    proofSize: '500000',
  })

  return sourceClient.api.tx.utility.forceBatch([xcmTx])
}

/**
 * Simulates and executes the full governance workflow to authorize a runtime upgrade
 * on a target chain, optionally across chains using collectives and XCM.
 *
 * - Considering any existing authorized upgrade
 * - Submitting the upgrade call preimage
 * - Attempt to dispatch when call is not yet whitelisted
 * - Whitelisting the call via a Collectives chain
 * - Attempting to dispatch with both valid and invalid origins.
 * - Validating that the upgrade was correctly authorized on the target chain.
 *
 * @param governingChain - The chain where the governance is running, allowed to execute as superuser on other chains
 * @param chainToUpgrade - The chain whose runtime is being upgraded.
 * @param collectivesChain - The chain that hosts the collective body which will whitelist the upgrade call.
 */
export async function authorizeUpgradeViaCollectives(
  governingChain: Client,
  chainToUpgrade: Client,
  collectivesChain: Client,
) {
  // NOTE: Since the test is run against some live chain data, it may happen that at some moment some upgrade
  //       is already authorized - expected result is that the authorized hash will be overriden by this test
  //
  // specific value of codeHash is not important for the test
  const codeHash = '0x0101010101010101010101010101010101010101010101010101010101010101'
  const assertAuthorizedUpgradeUnchanged = async () => {
    const currentAuthorizedUpgrade = await chainToUpgrade.api.query.system.authorizedUpgrade()
    assert(currentAuthorizedUpgrade.isNone || currentAuthorizedUpgrade.value.codeHash.toHex() !== codeHash)
  }
  await assertAuthorizedUpgradeUnchanged()

  const authorizeUpgradeCall =
    governingChain.url === chainToUpgrade.url
      ? chainToUpgrade.api.tx.system.authorizeUpgrade(codeHash)
      : createXcmAuthorizeUpgradeBatch(codeHash, governingChain, chainToUpgrade)

  const whiteListCall = governingChain.api.tx.whitelist.dispatchWhitelistedCallWithPreimage(
    authorizeUpgradeCall.method.toHex(),
  )

  const notePreimageTx = governingChain.api.tx.preimage.notePreimage(authorizeUpgradeCall.method.toHex())

  const okOrigin = { Origins: 'WhitelistedCaller' }
  // some existing but not allowed to dispatch whitelisted call origin
  const badOrigin = { Origins: 'StakingAdmin' }

  // it does not matter who is the origin which notes the preimage
  const noteEvents = await sendTransaction(notePreimageTx.signAsync(defaultAccounts.alice))
  await governingChain.dev.newBlock()
  await checkEvents(noteEvents, 'preimage').toMatchSnapshot('events after notePreimge')

  // try to dispatch a call that has not yet been whitelisted - should result in dispatching error
  await scheduleInlineCallWithOrigin(governingChain, whiteListCall.method.toHex(), okOrigin)
  await governingChain.dev.newBlock()
  await checkSystemEvents(governingChain, 'scheduler')
    .redact({ hash: false, redactKeys: /task/ })
    .toMatchSnapshot('events when dispatching non-whitelisted call')
  await assertAuthorizedUpgradeUnchanged()

  // collectives whitelisting a call
  await sendWhitelistCallViaXcmTransact(governingChain, collectivesChain, authorizeUpgradeCall.method.hash.toHex(), {
    proofSize: '10000',
    refTime: '500000000',
  })
  await collectivesChain.dev.newBlock()
  await checkSystemEvents(collectivesChain, 'polkadotXcm')
    .redact({ hash: false, redactKeys: /messageId/ })
    .toMatchSnapshot('collectives events emitted when sending xcm')
  await governingChain.dev.newBlock()
  await checkSystemEvents(governingChain, 'whitelist', 'messageQueue')
    .redact({ hash: false, redactKeys: /id/ })
    .toMatchSnapshot('governing chain events emitted on receiving xcm from collectives')

  // trying to dispatch whitelisted call using bad origin - should result in error
  await scheduleInlineCallWithOrigin(governingChain, whiteListCall.method.toHex(), badOrigin)
  await governingChain.dev.newBlock()
  await checkSystemEvents(governingChain, 'scheduler')
    .redact({ hash: false, redactKeys: /task/ })
    .toMatchSnapshot('events when dispatching whitelisted call with bad origin')
  await assertAuthorizedUpgradeUnchanged()

  // call is whitelisted, origin is ok - success expected
  await scheduleInlineCallWithOrigin(governingChain, whiteListCall.method.toHex(), okOrigin)
  await governingChain.dev.newBlock()
  await checkSystemEvents(governingChain, 'whitelist')
    .redact({ hash: false })
    .toMatchSnapshot('governing chain events about dispatching whitelisted call')

  if (governingChain.url === chainToUpgrade.url) {
    await checkSystemEvents(chainToUpgrade, { section: 'system', method: 'UpgradeAuthorized' })
      .redact({ hash: false })
      .toMatchSnapshot('to-be-upgraded chain events to confirm authorized upgrade')
  } else {
    await chainToUpgrade.dev.newBlock()
    await checkSystemEvents(chainToUpgrade, 'messageQueue', { section: 'system', method: 'UpgradeAuthorized' })
      .redact({ hash: false, redactKeys: /id/ })
      .toMatchSnapshot('to-be-upgraded chain events to confirm authorized upgrade')
  }

  assert.equal((await chainToUpgrade.api.query.system.authorizedUpgrade()).value.codeHash.toHex(), codeHash)
}
