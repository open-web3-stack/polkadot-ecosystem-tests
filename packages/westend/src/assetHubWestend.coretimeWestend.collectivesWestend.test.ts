import { describe, test } from 'vitest'

import { assetHubWestend, collectivesWestend, coretimeWestend } from '@e2e-test/networks/chains'
import { setupNetworks } from '@e2e-test/shared'
import { authorizeUpgradeViaCollectives } from '@e2e-test/shared/upgrade.js'

describe('assetHubWestend & coretimeWestend & collectivesWestend', async () => {
  const [ahClient, cortetimeClient, collectivesClient] = await setupNetworks(
    assetHubWestend,
    coretimeWestend,
    collectivesWestend,
  )

  test('Asset Hub authorizes People upgrade via Collectives', async () => {
    await authorizeUpgradeViaCollectives(ahClient, cortetimeClient, collectivesClient)
  })
})
