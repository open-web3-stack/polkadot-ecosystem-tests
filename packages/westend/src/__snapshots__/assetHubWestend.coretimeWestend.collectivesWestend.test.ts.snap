// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`assetHubWestend & coretimeWestend & collectivesWestend > Asset Hub authorizes People upgrade via Collectives > collectives events emitted when sending xcm 1`] = `
[
  {
    "data": {
      "destination": {
        "interior": {
          "X1": [
            {
              "Parachain": 1000,
            },
          ],
        },
        "parents": 1,
      },
      "message": [
        {
          "UnpaidExecution": {
            "checkOrigin": null,
            "weightLimit": "Unlimited",
          },
        },
        {
          "Transact": {
            "call": {
              "encoded": "0x5d0020883d25f1db1e4a6789b18726a0e94c8a8824c7f329d010633c5d8ad9a1a730",
            },
            "fallbackMaxWeight": {
              "proofSize": 10000,
              "refTime": 500000000,
            },
            "originKind": "Xcm",
          },
        },
      ],
      "messageId": "(redacted)",
      "origin": {
        "interior": {
          "X1": [
            {
              "Plurality": {
                "id": "Technical",
                "part": "Voice",
              },
            },
          ],
        },
        "parents": 0,
      },
    },
    "method": "Sent",
    "section": "polkadotXcm",
  },
]
`;

exports[`assetHubWestend & coretimeWestend & collectivesWestend > Asset Hub authorizes People upgrade via Collectives > events after notePreimge 1`] = `
[
  {
    "data": {
      "hash_": "(hash)",
    },
    "method": "Noted",
    "section": "preimage",
  },
]
`;

exports[`assetHubWestend & coretimeWestend & collectivesWestend > Asset Hub authorizes People upgrade via Collectives > events when dispatching non-whitelisted call 1`] = `
[
  {
    "data": {
      "id": null,
      "result": {
        "Err": {
          "Module": {
            "error": "0x03000000",
            "index": 93,
          },
        },
      },
      "task": "(redacted)",
    },
    "method": "Dispatched",
    "section": "scheduler",
  },
]
`;

exports[`assetHubWestend & coretimeWestend & collectivesWestend > Asset Hub authorizes People upgrade via Collectives > events when dispatching whitelisted call with bad origin 1`] = `
[
  {
    "data": {
      "id": null,
      "result": {
        "Err": "BadOrigin",
      },
      "task": "(redacted)",
    },
    "method": "Dispatched",
    "section": "scheduler",
  },
]
`;

exports[`assetHubWestend & coretimeWestend & collectivesWestend > Asset Hub authorizes People upgrade via Collectives > governing chain events about dispatching whitelisted call 1`] = `
[
  {
    "data": {
      "callHash": "0x20883d25f1db1e4a6789b18726a0e94c8a8824c7f329d010633c5d8ad9a1a730",
      "result": {
        "Ok": {
          "actualWeight": {
            "proofSize": "(rounded 3600)",
            "refTime": "(rounded 120000000)",
          },
          "paysFee": "Yes",
        },
      },
    },
    "method": "WhitelistedCallDispatched",
    "section": "whitelist",
  },
]
`;

exports[`assetHubWestend & coretimeWestend & collectivesWestend > Asset Hub authorizes People upgrade via Collectives > governing chain events emitted on receiving xcm from collectives 1`] = `
[
  {
    "data": {
      "callHash": "0x20883d25f1db1e4a6789b18726a0e94c8a8824c7f329d010633c5d8ad9a1a730",
    },
    "method": "CallWhitelisted",
    "section": "whitelist",
  },
  {
    "data": {
      "id": "(redacted)",
      "origin": {
        "Sibling": "(rounded 1000)",
      },
      "success": true,
      "weightUsed": {
        "proofSize": "(rounded 3600)",
        "refTime": "(rounded 120000000)",
      },
    },
    "method": "Processed",
    "section": "messageQueue",
  },
]
`;

exports[`assetHubWestend & coretimeWestend & collectivesWestend > Asset Hub authorizes People upgrade via Collectives > to-be-upgraded chain events to confirm authorized upgrade 1`] = `
[
  {
    "data": {
      "checkVersion": true,
      "codeHash": "0x0101010101010101010101010101010101010101010101010101010101010101",
    },
    "method": "UpgradeAuthorized",
    "section": "system",
  },
  {
    "data": {
      "id": "(redacted)",
      "origin": {
        "Sibling": 1000,
      },
      "success": true,
      "weightUsed": {
        "proofSize": 0,
        "refTime": "(rounded 120000000)",
      },
    },
    "method": "Processed",
    "section": "messageQueue",
  },
]
`;
